### PR TITLE
[aptos-rosetta] Add network & block commands to rosetta

### PR DESCRIPTION
**Pull request by gregnazario**
## Motivation

This builds the building blocks for Rosetta, along with CLI to be able to test them easier than the Rosetta CLI.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

See E2E tests & CLI

## Related PRs

This builds upon https://github.com/aptos-labs/aptos-core/pull/1243

### CODE REVIEW

File: crates/aptos-rosetta/src/block.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    warp::post().and(
+        warp::path!("block")
+            .and(warp::body::json())
+            .and(with_context(server_context))
+            .and_then(handle_request(block)),
+    )
+}
+
+async fn block(request: BlockRequest, server_context: RosettaContext) -> ApiResult<BlockResponse> {
+    debug!("/block");
```
Comment (davidiw) [inline crates/aptos-rosetta/src/block.rs:29]: trace log?

**Review by davidiw — COMMENTED** [review]
Why are we introducing our own CLI, when this exists? All I saw was "easier". But it is also correct right? How are we evaluating this for correctness?
Also why introduce the types that aren't used in this PR?

Other than that, looks good!

File: crates/aptos-rosetta/src/block.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos-rosetta/src/block.rs:29]: Yeah, missed the trace logs, will add them.

**Comment by gregnazario** [comment]
> Why are we introducing our own CLI, when this exists? All I saw was "easier". But it is also correct right? How are we evaluating this for correctness?
> 
> Other than that, looks good!

I don't really want my own CLI, but it will still run the correctness via the Rosetta version.

This lets me quickly evaluate happy case against devnet with the actual Rosetta server including flows.  I can delete it later.

> Why introduce types not used in this PR?

This is because I took the strategy of just building all the object required from the spec, and then implementing APIs.  The only well defined parts of the spec are the types, but the behavior isn't well described.

**Review by davidiw — APPROVED** [review]

**Pull request merged**